lib/src/solver/report.dart

```diff
@@ -130,6 +130,33 @@ class SolveReport {
     }
   }
 
+  /// Displays a two-line message, number of outdated packages and an
+  /// instruction to run `pub outdated` if outdated packages are detected.
+  void reportOutdated() {
+    final outdatedPackagesCount = _result.packages.where((id) {
+      final versions = _result.availableVersions[id.name];
+      // A version is counted:
+      // - if there is a newer version which is not a pre-release and current
+      // version is also not a pre-release
+      // - if the current version is pre-release then any upgraded version is
+      // considered.
+      return versions.any((v) =>
+          v > id.version &&
+          (id.version.isPreRelease == true || v.isPreRelease == false));
```

Comment:
Nit: I think this line could be simplified:
```
 (id.version.isPreRelease || !v.isPreRelease));
```

Reply:
Done!